README.md

```diff
@@ -165,6 +165,7 @@ If the caller does not want to be blocked, then he could call the function insid
 	go func(c chan fetchResult) {
 		c <- fetchA()
 	}(aChan)
+	a := <-aChan
```

Comment:
The thing here is that this line is effectively "blocking".
It can demonstrate concurrency when in the presence of other channels as the example here https://github.com/pazams/go-for-javascript-developers#d-sequential-and-concurrent-patterns ("order doesn't matter" comment).

We can perhaps extend this example to include a `select` with an undefined `otherChan`, or have it mentioned in a separated following snippet. 

Let's keep the discussion rolling on this one.

Reply:
my current knowledge of Go ends here ;)

Reply:
No worries! thanks for bringing this into attention :)

Reply:
I read about channels in detail. I think we should have this line `a := <-aChan`, because although this line is blocking. This blocking is not due to the `fetchA()` function. In synchronous side, the `fetchA()` function is called immediately, however in the asynchronous side, the `fetchA()` function is not called immediately. We can do anything after defining the goroutine func literal and immediately calling it. When we have nothing else to do then we can start listening to the channel like so `a := <-aChan`. At this point the call is waiting. In conclusion I think we should do this snippet
```Go
 	aChan := make(chan fetchResult, 0)
 	go func(c chan fetchResult) {
 		c <- fetchA()
 	}(aChan)
        fmt.Println("Do anything")
	a := <-aChan

```

Reply:
hi @pazams what are your thoughts about this

Reply:
@pazams I just made a PR for using `select`, check out #21